services/app-web/src/gqlHelpers/fetchHealthPlanPackageWrapper.ts

```diff
@@ -0,0 +1,191 @@
+function wrapApolloResult({
+    loading,
+    error,
+    data,
+}: {
+    loading: boolean
+    error?: ApolloError
+    data: FetchHealthPlanPackageQuery | undefined
+}): FetchHealthPlanPackageResult {
+    if (loading) {
+        return {
+            status: 'LOADING',
+        }
+    }
+
+    if (error) {
+        return {
+            status: 'ERROR',
+            error: error,
+        }
+    }
+
+    if (data) {
+        return {
+            status: 'SUCCESS',
+            data,
+        }
+    }
+
+    return {
+        status: 'ERROR',
+        error: new Error('UNEXPECTED APOLLO BEHAVIOR, NO DATA'),
+    }
+}
+
+// This is ugly, but incredibly useful.
+// We want to hide our protobuf parsing from our pages. The GQL
+// query returns, several layers down, HealthPlanRevisions which have
+// a formDataProto property on it holding the encoded proto string.
+// We want our wrapper to return the identical query to graphql, but
+// with all the HealthPlanRevisions having an additional `formData`
+// property that contains the parsed HealthPlanFormDataType
+// domain model.
+
+// To start, we construct our ParsedHealthPlanRevision by extending
+// that type with the `formData` property.
+interface ParsedHealthPlanRevision extends HealthPlanRevision {
+    formData: HealthPlanFormDataType
+}
+
+// Then, we recursively replace each property in the chain from there to the
+// query with the existing type, removing the one containing type and replacing it
+// with our parsed type.
+// So, here, we take the HealthPlanRevisionEdge, remove `node` and extend it with
+// our parsed `node`
+interface ParsedHealthPlanRevisionEdge
+    extends Omit<HealthPlanRevisionEdge, 'node'> {
+    node: ParsedHealthPlanRevision
+}
+interface ParsedPackageType extends Omit<HealthPlanPackage, 'revisions'> {
+    revisions: ParsedHealthPlanRevisionEdge[]
+}
+interface ParsedFetchHealthPlanPackageSubQuery
+    extends Omit<FetchHealthPlanPackageQuery['fetchHealthPlanPackage'], 'pkg'> {
+    pkg?: ParsedPackageType | null
+}
+interface ParsedFetchHealthPlanPackageQuery
+    extends Omit<FetchHealthPlanPackageQuery, 'fetchHealthPlanPackage'> {
+    fetchHealthPlanPackage: ParsedFetchHealthPlanPackageSubQuery
+}
+
+type ParsedFetchResult =
+    | Exclude<FetchHealthPlanPackageResult, FetchSuccess>
+    | {
+          status: 'SUCCESS'
+          data: ParsedFetchHealthPlanPackageQuery
+      }
+
+// This wraps our call to useFetchHealthPlanPackageQuery, parsing out the protobuf
+// from the response, returning extra errors in the case that parsing goes wrong
+function useFetchHealthPlanPackageWrapper(id: string): ParsedFetchResult {
+    const result = wrapApolloResult(
+        useFetchHealthPlanPackageQuery({
+            variables: {
+                input: {
+                    pkgID: id,
+                },
+            },
+        })
+    )
+
+    if (result.status === 'SUCCESS') {
+        const pkg = result.data.fetchHealthPlanPackage.pkg
```

Comment:
💡 No action needed but a thought - This `parsedPackage` code seems like the most extendable part of the wrapper. It seems relevant for any query... now or in future...  that would return a health plan package with a submission and revisions. At quick glance though one issue is we have no real type for ParsedPackageType.  interesting problem to poke at in the future